core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -677,10 +682,35 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   /**
    * Replay the given log file, saving the application in the listing db.
    */
+
   protected def mergeApplicationListing(
       fileStatus: FileStatus,
       scanTime: Long,
       enableOptimizations: Boolean): Unit = {
+    try {
+      mergeApplicationListingHelper(fileStatus, scanTime, enableOptimizations)
+    } catch {
+      case e: InterruptedException =>
```

Comment:
You may want to recheck all expressions in catch statement are thread-safe, for sure. (I don't know whether it's thread safe or not, but the logic is now executed from multiple threads concurrently while it was executed from single thread.)

Reply:
Hi, I have checked these expressions here and I think they are thread-safe.

- blacklist is a ConcurrentHashMap and blackList(path) is thread-safe.
- listing.delete() is thread-safe.
- endProcess is a remove operation for ConcurrentSkipListSet and it is thread-safe.
- pendingReplayTasksCount is an AtomicInteger and it is thread-safe.